### PR TITLE
Add AppVeyor integration

### LINKED ISSUE #31: Add Windows CI
In recent releases, `hsc2hs` has developed quite a few Windows-specific code paths, but since the current Travis CI setup only builds on Linux, these code paths are not exercised on CI. We should set up AppVeyor CI to run on Windows, similarly to how the [`Win32` library's setup](https://ci.appveyor.com/project/hvr/win32) works.

### PR DESCRIPTION
**Pull request by RyanGlScott**
Let's get some Windows CI up in here.

Fixes #31.

### CODE REVIEW

**Comment by RyanGlScott** [comment]
It looks like that in order for GitHub to automatically build commits/PRs through AppVeyor, the following webhook must be added:

> https://ci.appveyor.com/api/github/webhook?id=01c9lgrksnptm41p 

I don't have the right permission levels to go into `hsc2hs`' settings and do this. Can someone else do this please?

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    - GHCVER: "7.6.3.1"
+
+cache:
+ - "C:\\SR"
+
+install:
+ - "choco install -y cabal"
+ - "choco install -y ghc --version %GHCVER%"
+ - "refreshenv"
+ - "set PATH=C:\\msys64\\mingw64\\bin;C:\\msys64\\usr\\bin;%PATH%"
```
Comment (Mistuke) [inline appveyor.yml]: You don't need this anymore. The package detects appveyor on its own now.

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    - GHCVER: "8.0.2.2"
+    - GHCVER: "7.10.3.2"
+    - GHCVER: "7.8.4.1"
+    - GHCVER: "7.6.3.1"
+
+cache:
+ - "C:\\SR"
+
+install:
+ - "choco install -y cabal"
```
Comment (Mistuke) [inline appveyor.yml]: You can drop the quotes around everything :) They make it a bit harder to read and when you do need quotes you'd need to double escape them

**Comment by Mistuke** [comment]
Thanks for adding this @RyanGlScott 

You may also want to consider adding
```
# Do not build feature branch with open Pull Requests
skip_branch_with_pr: true

matrix:
  # set this flag to immediately finish build once one of the jobs fails.
  fast_finish: true
```

the first one avoid building the same branch twice when someone submits a PR. and the second one just saves some compile time since you have a long list of GHCs in there. If one fails you want to stop as soon as possible I would imagine.

**Comment by hvr** [comment]
@RyanGlScott can you try rebasing this PR? (I did enable Appveyor for this repo)

File: appveyor.yml (hunk 1 above)
Comment (RyanGlScott) [reply appveyor.yml]: Good to know. I've dropped this line.

File: appveyor.yml (hunk 2 above)
Comment (RyanGlScott) [reply appveyor.yml]: Good advice—I've implemented it. In my defence, I copied this script from somewhere else a long while back, so I'm not surprised that this isn't adhering to best practices :)

**Comment by RyanGlScott** [comment]
Thanks for the helpful feedback, @Mistuke. And thanks for enabling AppVeyor support, @hvr. I just pushed and the AppVeyor build was automatically triggered, so it appears to have worked.

Speaking of which, it looks like the GitHub webhook is pointing to https://ci.appveyor.com/project/hvr/hsc2hs, but I have the `README` badge pointing at https://ci.appveyor.com/project/RyanGlScott/hsc2hs instead. We should probably just use your version everywhere, so @hvr could you tell me the URL for the AppVeyor badge for https://ci.appveyor.com/project/hvr/hsc2hs? (It should be under Settings > Badges > SVG branch image URL.)

**Comment by hvr** [comment]
@RyanGlScott https://ci.appveyor.com/api/projects/status/ee434vcpvit2qeqh?svg=true

However, if we add the badge to the README then we shouldn't include the README in the cabal package (which it currently isn't anyway, cause Hackage can't render .rst files *yet*)

**Comment by RyanGlScott** [comment]
> However, if we add the badge to the README then we shouldn't include the README in the cabal package (which it currently isn't anyway, cause Hackage can't render .rst files _yet_)

OK. Since we don't include the `README` in the `cabal` package currently, I suppose there's no issue with adding badges then? Or would badges make it more complicated for Hackage to render `.rst` files should `hackage-server` add support for doing so?

**Comment by hvr** [comment]
@RyanGlScott Yes, for now it's fine to include the badge; But I'd remove it again when the README gets included into the tarball at some point in the future; there's a conflict of interest/purpose w/ the READMEs as used on GitHub and those as used on Hackage; the ones on GitHub are supposed to be development centric; whereas the ones on Hackage are supposed to give more in depth explaination about a package that didn't fit into the indexed `description` field (see also [this posting](https://mail.haskell.org/pipermail/haskell-cafe/2016-November/125541.html) for more details).

While there might be packages that are unprincipled about this, for packages I maintain I try to give good examples and not conflate the two purposes of READMEs. One way to address this here might be to add a 2nd `README.md` which is intended for GitHub and leave the `README.rst` for its original scope; then GitHub will only render the README.md file by default.

**Comment by RyanGlScott** [comment]
https://github.com/haskell/hsc2hs/pull/32#issuecomment-537351096 sounds good to me. I'll go ahead and merge this, but we can always clean up the `README`(s) afterwards.

**Pull request merged**